### PR TITLE
fix: impersonation switch active state

### LINKED ISSUE #8907: Impersonation enable/disable toggle issue
The impersonation toggle button is not working as expected. When the button is clicked it should be enabled/disabled but it is working as mentioned in the screen recording below 

https://github.com/calcom/cal.com/assets/27686539/20ad2f98-72d6-4705-ad2d-3beb7020599a

If this is considered as an issue I can contribute to it

### PR DESCRIPTION
**Pull request by G3root**
## What does this PR do?

Fixes #8907 

after:

[Screencast 2023-05-16 14:26:23.webm](https://github.com/calcom/cal.com/assets/84864519/f0738974-b716-42cb-9d29-5b6ba6c7b6c4)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: apps/web/pages/settings/security/impersonation.tsx [hunk 1]
```diff
@@ -63,11 +64,11 @@ const ProfileImpersonationView = () => {
         }}>
         <div className="flex space-x-3">
           <Switch
-            defaultChecked={!user?.disableImpersonation}
             onCheckedChange={(e) => {
-              setValue("disableImpersonation", !e, { shouldDirty: true });
+              setValue("disableImpersonation", e, { shouldDirty: true });
```
Comment (sean-brydon) [inline apps/web/pages/settings/security/impersonation.tsx:68]: Are we meant to be removing !

This switch disables impersonation if set to true 

even tho the value coming from the switch is false 

false->disables impersonation
true(default)-> impersonation enabled

Comment (G3root) [reply apps/web/pages/settings/security/impersonation.tsx]: @sean-brydon pushed some changes

**Review by sean-brydon — APPROVED** [review]
Happy with this thanks!

**Review by alannnc — APPROVED** [review]
Looks good! My personal preference it's trigger the update on the toggle action and save 1 click, but can be handled later.

**Review by zomars — APPROVED** [review]

**Pull request merged**